lms/services/jwt.py

```diff
@@ -51,3 +56,40 @@ def encode_with_secret(cls, payload: dict, secret, lifetime):
         jwt_str = jwt.encode(payload, secret, algorithm="HS256")
 
         return jwt_str
+
+    @classmethod
+    def decode_unverified(cls, jwt_str):
+        """Decode a JWT token without any verification."""
+        return jwt.decode(jwt_str, options={"verify_signature": False})
+
+    @staticmethod
+    @lru_cache
+    def _get_jwk_client(jwk_url: str):
+        """
+        Get a PyJWKClient for the given key set URL.
+
+        PyJWKClient maintains a cache of keys it has seen we want to keep
+        the clients around with `lru_cache` in case we can reuse that internal cache
+        """
+        return jwt.PyJWKClient(jwk_url)
+
+    @classmethod
+    def decode_with_jwk_url(cls, jwt_str, jwk_url: str, audience: str):
+        """
+        Decode a JWK verifying it against the public key published at `jwk_url`.
+
+        The "aud" claim of the JWT must match the `audience` parameter.
+        """
+        header = jwt.get_unverified_header(jwt_str)
+        if not header.get("kid"):
+            LOG.debug("Error decoding JWT. No `kid` in header")
+            raise InvalidJWTError("Missing 'kid' value in JWT header")
```

Comment:
The exception will be handled and won't show up in the logs.

We are asserting the logging in the test and it's also useful for the certification process: https://github.com/hypothesis/lms/pull/3835/files#r844832637